src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py

```diff
@@ -294,6 +294,24 @@ def set_attrs(self, path, attrs):
             fs_earmark = CephFSVolumeEarmarking(self.fs, path)
             fs_earmark.set_earmark(earmark)
 
+        case_sensitive = attrs.get("case_sensitive")
+        if case_sensitive is not None:
+            if case_sensitive not in ("0", "1"):
+                raise VolumeException(-errno.EINVAL, "invalid case sensitivity")
+            try:
+                self.fs.setxattr(path, "ceph.dir.casesensitive", case_sensitive.encode('utf-8'), 0)
+            except cephfs.Error as e:
+                raise VolumeException(-e.args[0], e.args[1])
+
+        normalization = attrs.get("normalization")
+        if normalization is not None:
+            if normalization not in ("nfd", "nfc", "nfkd", "nfkc"):
+                raise VolumeException(-errno.EINVAL, "invalid unicode normalization form")
```

Comment:
Doesn't `setxattr` give you `-EINVAL` if you give an invalid value? Why also check here?

Reply:
Done